### PR TITLE
PP 51 notifications socket io

### PR DESCRIPTION
**Pull request by mariyador**
### Description
This update includes installation of socket functionality for real-time user connection management, and the addition of approval/rejection features for class applications with real-time notifications.
### Related issues

- Socket Installation: Socket functionality was installed to manage real-time connections, ensuring users are notified promptly about relevant actions.
- Notification for Class Applications: An event emitter was added to notify class creators when a new application is submitted for their class.
- Approval/Reject Functionality: Added functionality to approve or reject class applications, including emitting notifications to applicants upon decision.

### CODE REVIEW

**Review by ofeterovskaya — APPROVED** [review]
You did fantastic job, Maria, everything looks great on my side

File: src/app.js [hunk 1]
```diff
@@ -46,4 +51,10 @@ app.use('/api/v1/classes', classesRouter);
 app.use(notFound);
 app.use(errorHandlerMiddleware);
 
-module.exports = app;
+const server = http.createServer(app);
+const io = new Server(server, {
+  cors: { origin: 'http://localhost:5173', method: ['GET', 'POST'] },
```
Comment (AnnaPestova1) [inline src/app.js:56]: Hi Maria! We already deployed our project, don't you mind to add the additional origin linked the app's url?

File: src/controllers/classesController.js [hunk 2]
```diff
@@ -37,21 +37,30 @@ const displaySearchClasses = async (req, res) => {
...
+      const total = await Class.countDocuments(query);
+
+      res.status(StatusCodes.OK).json({
+        classes,
+        total,
+        totalPages: Math.ceil(total / limit),
+        currentPage: page,
+      });
+    } else {
+      const classes = await Class.find().sort({ [sortBy]: sortOrder });
```
Comment (AnnaPestova1) [inline src/controllers/classesController.js:55]: is this intentional change of logic - to show all classes on the one page if no search query?

File: src/controllers/classesController.js [hunk 3]
```diff
@@ -80,11 +89,17 @@ const getClassDetails = async (req, res) => {
...
 };
 
-//Create a class, only after login
 const createClass = async (req, res) => {
-  const createdBy = req.user.userId;
+  const { userId, role } = req.user;
 
   try {
+    // Check if the user has the role of a teacher
+    if (role !== 'teacher') {
```
Comment (AnnaPestova1) [inline src/controllers/classesController.js:97]: Great addition to the code

File: src/controllers/classesController.js (hunk 2 above)
Comment (mariyador) [reply src/controllers/classesController.js:55]: Yes, but I planned to rewrite displaySearchClasses in a separate PR to improve our search functionality.

File: src/app.js (hunk 1 above)
Comment (mariyador) [reply src/app.js]: Thank you, Anna!
✅Done

**Review by AnnaPestova1 — APPROVED** [review]
Great work Maria! 
Very clear code and good improvements as always.
Just quick thought: it would be awesome if our app would have the cors not only for sockets, but for all app.

**Comment by mariyador** [comment]
> Great work Maria! Very clear code and good improvements as always. Thank you for quick response on my messages for this PR. Just quick thought: it would be awesome if our app would have the cors not only for sockets, but for all app.

Thank you, Anna!
I will do it in separate PR

**Pull request merged**